src/event.h

```diff
@@ -10,64 +10,132 @@
+    // no copying or moving of any kind (for simplicity and to prevent accidental copies)
+    ev_entry(ev_entry &&) = delete;
 
-    int weight;
-    bool cbdata;
+public:
+    double when; ///< when this event is due to run
+    int weight; ///< deprecated weight hack, avoids
+    AsyncCall::Pointer call; ///< the AsyncCall which will perform this event
 
-    ev_entry *next;
+    ev_entry *next = nullptr;
 };
 
-// manages time-based events
+/// Manages time-based events
 class EventScheduler : public AsyncEngine
 {
 
 public:
     EventScheduler();
     ~EventScheduler();
-    /* cancel a scheduled but not dispatched event */
-    void cancel(EVH * func, void * arg);
-    /* clean up the used memory in the scheduler */
+
+    /// clean up the used memory in the scheduler
     void clean();
-    /* either EVENT_IDLE or milliseconds remaining until the next event */
+
+    /// either EVENT_IDLE or milliseconds remaining until the next event
     int timeRemaining() const;
-    /* cache manager output for the event queue */
+
+    /// cache manager output for the event queue
     void dump(Packable *);
-    /* find a scheduled event */
-    bool find(EVH * func, void * arg);
-    /* schedule a callback function to run in when seconds */
+
+    /// schedule an AsyncCall run in when seconds
+    void schedule(const AsyncCall::Pointer &, double when, int = 0);
+    /// schedule an AsyncCall run in when +/-30% seconds
+    void scheduleIsh(const AsyncCall::Pointer &, double when);
+    /// \deprecated schedule a callback function to run in when seconds. Use AsyncCall instead.
     void schedule(const char *name, EVH * func, void *arg, double when, int weight, bool cbdata=true);
-    int checkEvents(int timeout);
-    static EventScheduler *GetInstance();
+
+    /// \deprecated find a scheduled callback function. Use AsyncCall instead.
+    bool find(EVH * func, void * arg);
+
+    /// \deprecated cancel a scheduled but not dispatched event
+    /// Use AsyncCall API to cancel() the Call instead.
+    void cancel(EVH * func, void * arg);
+
+    /// remove a scheduled but not dispatched event
+    void remove(const AsyncCall::Pointer &);
+
+    /* AsyncEngine API */
+    int checkEvents(int) override;
 
 private:
-    static EventScheduler _instance;
     ev_entry * tasks;
 };
 
+/// global queue of events waiting to happen
+EventScheduler &Events();
+
+/// perform any startup tasks needed by the EventScheduler
+extern void eventInit();
+
+/// \see EventScheduler::clean()
+inline void
+eventFreeMemory()
+{
+    Events().clean();
+}
+
+/// \see EventScheduler::schedule()
+inline void
+eventAdd(AsyncCall::Pointer &c, double when)
+{
+    Events().schedule(c, when);
+}
+
+/// \see EventScheduler::scheduleIsh()
+inline void
+eventAddIsh(AsyncCall::Pointer &c, double when)
+{
+    Events().scheduleIsh(c, when);
+}
+
+/// \see EventScheduler::remove()
+inline void
+eventDelete(AsyncCall::Pointer &c)
+{
+    Events().remove(c);
+}
```

Comment:
I see you missed the irony. This *is* the old API, it is just taking the new parameter type (Also why I left the new-API callers using the new API methods instead of the old API C functions.)

> 
> The same applies to other new AsyncCall-based functions, of course. Suggestions: ScheduleEvent() and ScheduleEventAtApproximateTime() or ScheduleEventWithJitter() or even ScheduleWobblyEvent().

Ew. Please avoid the COBOL coding style guides, paraticularly if you are going to insist on Americanized grammar for wording.

This is C++ we are writing. It has a grammar syntactic like:  Within classDo'er perform methodActions using Params    -> EventScheduler().remove(c)

Reply:
> This _is_ the old API

My change request is about the name of the `eventDelete(AsyncCall::Pointer &c)` function. That function is new. We can, and should, name it differently from the legacy eventDelete() function (which does not work with AsyncCalls and has many serious problems).

> Ew. Please avoid the COBOL coding style guides.

> This is C++ we are writing. It has a grammar syntactic like: Within classDo'er perform methodActions using Params -> EventScheduler().remove(c)

I am not sure why you are saying all this. My change request is about the name of the stand-alone _function_, not anything "within" a class.